addons/mrp/tests/test_bom.py

```diff
@@ -1961,3 +1961,75 @@ def test_component_when_bom_change(self):
         # Check that component 1 is set again and component 2 is removed
         self.assertEqual(mo_form.move_raw_ids._records[0]['product_id'], self.product_1.id)
         self.assertEqual(len(mo_form.move_raw_ids._records), 1)
+
+    def test_update_operations(self):
+        """Update the operations in BoM which reflects the changes in Manufacturing Order"""
+
+        test_bom = self.env['mrp.bom'].create({
+            'product_tmpl_id': self.product_2.product_tmpl_id.id,
+            'product_uom_id': self.product_2.uom_id.id,
+            'product_qty': 1.0,
+            'type': 'normal',
+            'bom_line_ids': [
+                (0, 0, {'product_id': self.product_7_1.id, 'product_qty': 1.0}),
+            ],
+            'operation_ids': [
+                Command.create({
+                    'name': 'Cutting',
+                    'workcenter_id': self.workcenter_1.id,
+                    'sequence': 1
+                })
+            ],
+        })
+        mo_form = Form(self.env['mrp.production'].with_user(self.user_mrp_user))
+        mo_form.product_id = self.product_7_1
+        mo_form.product_qty = 1.0
+        mo_form.bom_id = test_bom
+        mo = mo_form.save()
+        mo.action_confirm()
+
+        test_bom.operation_ids.update({
+            'name': 'Painting',
+            'workcenter_id': self.workcenter_2.id
+        })
+
+        mo.action_update_bom()
+        self.assertEqual(test_bom.operation_ids.name, mo.bom_id.operation_ids.name)
+        self.assertEqual(test_bom.operation_ids.workcenter_id, mo.bom_id.operation_ids.workcenter_id)
```

Comment:
you are testing the same records here `test_bom == mo.bom_id`... You need to test the `mo.workorder_ids`